blockchain/blockchain-wallet-v4-frontend
# test - login saga tests

## Description
Unit tests for sagas. Redux-saga-test-plan used to save some space.
Notice that the case where session is selected is tested using the "integrational" approach: yield order does not matter, but instead you can test only required yields. Can slow down tests time if such approach is abused(e.g. every single yield is tested by it)

## Change Type
- Upgrades

## Testing Steps
1. Check out slight upgrade to the error handling: previous version would throw on non-object errors.
2. Check out integrational approach( you can find it by searching obtainSessionToken) and compare it against more-concise step-by step dive into saga with saves and restores.

## Code Checklist
- [+] Code compiles successfully (verified via `yarn start`)
- [+] No lint issues exist (verified via `yarn lint`)
- [+] New and existing unit tests pass (verified via `yarn test`)
- [+] `README.md` and other documentation is updated as needed
